docs/adr/0008-managementv3-clusters-support.md

```diff
@@ -0,0 +1,49 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [8. Management V3 clusters support](#8-management-v3-clusters-support)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 8. Management V3 clusters support
+
+* Status: proposed
+* Date: 2024-01-25
+* Authors: @salasberryfin
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+As an operator, I'd like to have an alternative import strategy that will generate a `management.cattle.io/v3` cluster instead of a `provisioning.cattle.io/v1` cluster. This is to ensure that we can support this scenario in light of future changes to Rancher.
+
+## Decision
+
+The current import controller logic creates a `provisioning.cattle.io/v1` cluster when a new CAPI cluster is detected. To support a new import strategy, based on a new feature flag, Rancher Turtles will either start the existing `provisioning.cattle.io` controller or a new one for `management.cattle.io`.
+
+Adding support for this resource presents some challenges:
+- Owner Reference cannot be set: `management.cattle.io/v3` clusters are cluster-scoped while CAPI clusters are namespaced.
+- Display name and resource name are different: the resource name must satisfy the regex `^(c-[a-z0-9]{5})$`.
```

Comment:
The regex is set by Rancher [here](https://github.com/rancher/rancher/blob/676c3eb3dfa5a5d7d359315f942b630f8d3bf479/pkg/controllers/provisioningv2/cluster/controller.go#L50) and only clusters that satisfy it will be shown in the dashboard. I'd say it is relevant to mention its current state as a reason why the decision was made. I think that it would be best that, if it were to change in the future, we create a new document or simply update the code accordingly.

Reply:
Rancher has a similar [issue](https://github.com/rancher/rancher/issues/40852) for the regex. As I said, it’s enough to say that we use `generateName: c-` for cluster creation. This will cover it, and we won’t propagate the problem further in a design document. It’s not a good practice to rely on internal functionality prone to change, and mention it in a design documents for the same reason.

Reply:
References to the current regular expression are now removed from the document.